### PR TITLE
Change the arg names and make parameters more meaningful

### PR DESCRIPTION
**Pull request by liukun4515**
# Which issue does this PR close?

Before refactor the coercion framework, I take a look some  cal `result_type` or `create_xx_phy_expr`, and find some parameters has no meaningful. It's very easy to make mistakes.

related #1356
related #1358 

 # Rationale for this change

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/src/physical_plan/aggregates.rs [hunk 1]
```diff
@@ -93,90 +93,111 @@ impl FromStr for AggregateFunction {
...
         return Err(DataFusionError::Plan(format!(
             "Invalid or wrong number of arguments passed to aggregate: '{}'",
             name,
         )));
     }
-    let arg = arg[0].clone();
+    let first_coerced_phy_expr = coerced_phy_exprs[0].clone();
 
-    let arg_types = args
+    let coerced_exprs_types = coerced_phy_exprs
```
Comment (liukun4515) [inline datafusion/src/physical_plan/aggregates.rs:143]: this is the point about the #1358 issued by me.

**Comment by liukun4515** [comment]
@alamb @Dandandan  @houqp  PTAL

File: datafusion/src/physical_plan/aggregates.rs [hunk 2]
```diff
@@ -93,90 +93,111 @@ impl FromStr for AggregateFunction {
...
 
-/// Create a physical (function) expression.
-/// This function errors when `args`' can't be coerced to a valid argument type of the function.
+/// Create a physical aggregation expression.
+/// This function errors when `input_phy_exprs`' can't be coerced to a valid argument type of the aggregation function.
 pub fn create_aggregate_expr(
     fun: &AggregateFunction,
     distinct: bool,
-    args: &[Arc<dyn PhysicalExpr>],
+    input_phy_exprs: &[Arc<dyn PhysicalExpr>],
```
Comment (alamb) [inline datafusion/src/physical_plan/aggregates.rs:129]: I happen to think `input_phy_exprs` is more clear and I prefer it, however the rest of the code uses `arg` or `args` pretty extensively to refer to the input arguments of functions/aggregates/ etc as `args` 

I do like how the difference between `Expr` and `PhysicalExpr` is now clearer to the reader without having to look at the definition

**Review by alamb — APPROVED** [review]
Thank you @liukun4515  -- I happen to think this is an improvement in readability. 

I would like to leave it open for a little while to see what opinion other reviewers might have.

File: datafusion/src/physical_plan/aggregates.rs (hunk 1 above)
Comment (liukun4515) [reply datafusion/src/physical_plan/aggregates.rs:143]: @alamb  could you please take a look this point?

File: datafusion/src/physical_plan/aggregates.rs (hunk 2 above)
Comment (liukun4515) [reply datafusion/src/physical_plan/aggregates.rs:129]: Yes, i think this changes just make the code clearly.
Reader and reviewer can more easily find out the different point.

**Comment by liukun4515** [comment]
> Thank you @liukun4515 -- I happen to think this is an improvement in readability.
> 
> I would like to leave it open for a little while to see what opinion other reviewers might have.

thanks for your review.
There are some files with the `args` or `arg` except these files.

File: datafusion/src/physical_plan/aggregates.rs (hunk 1 above)
Comment (houqp) [reply datafusion/src/physical_plan/aggregates.rs:143]: good catch on this :+1:

**Review by houqp — APPROVED** [review]
personally i find the orginal `arg` naming quite readable for me because I consider arg the same as function input. but if you all think input reads better, then it's a +1 from me :) I do think the `coerced_` prefix improved readability for me.

File: datafusion/src/physical_plan/aggregates.rs [hunk 3]
```diff
@@ -93,90 +93,111 @@ impl FromStr for AggregateFunction {
...
-            Arc::new(expressions::Min::new(arg, name, return_type))
-        }
-        (AggregateFunction::Max, _) => {
-            Arc::new(expressions::Max::new(arg, name, return_type))
-        }
-        (AggregateFunction::Avg, false) => {
-            Arc::new(expressions::Avg::new(arg, name, return_type))
+        (AggregateFunction::ApproxDistinct, _) => {
+            Arc::new(expressions::ApproxDistinct::new(
+                first_coerced_phy_expr,
```
Comment (houqp) [inline datafusion/src/physical_plan/aggregates.rs:176]: not related to your change, just a note for future refactoring. I think it's more readable to use `coerced_phy_exprs[0].clone()` here directly and get rid of the declaration of `first_coerced_phy_expr`. Also makes it consistent with the `coerced_exprs_types[0].clone()` below.

File: datafusion/src/physical_plan/aggregates.rs (hunk 1 above)
Comment (liukun4515) [reply datafusion/src/physical_plan/aggregates.rs:143]: @houqp Maybe i made a mistake.
In the expression calculation, the returned data type should be calc from original input types and function.
The original logic is right. 
https://github.com/apache/arrow-datafusion/blob/15db29153aefc9b759ec03e6eeab8cb9ebf87781/datafusion/src/physical_plan/functions.rs#L1144
this is is my intention.

File: datafusion/src/physical_plan/aggregates.rs (hunk 3 above)
Comment (liukun4515) [reply datafusion/src/physical_plan/aggregates.rs:176]: can't agree more

**Review by alamb — APPROVED** [review]
I think our tests cover this pretty well. Thanks @liukun4515

File: datafusion/src/physical_plan/aggregates.rs (hunk 1 above)
Comment (houqp) [reply datafusion/src/physical_plan/aggregates.rs:142]: ha, ok, so what we had in `physical_plan/functions.rs` was wrong :D

**Pull request merged**